BFE_RShiny/oasisui/R/buildFly_module.R

```diff
@@ -0,0 +1,527 @@
+    result$settings_df[inp_celledit$row, "changed"] <- TRUE
+  })
+
+  # dynamically create fileInputs and DTOutputs
+  output$browsers_tables <- renderUI({
+    df_selectors <- result$tbl_modelsDetails$data_settings$datafile_selectors
+    if (!is.null(df_selectors)) {
+      result$inputID <- lapply(seq_len(length(df_selectors)), function(i) {
+        df_selectors[[i]]$name
+      })
+      label_pre <- lapply(seq_len(length(df_selectors)), function(i) {
+        paste(strsplit(df_selectors[[i]]$name, "_")[[1]], collapse = " ")
+      })
+      accept <- lapply(seq_len(length(df_selectors)), function(i) {
+        ext <- paste0(".", unlist(df_selectors[[i]]$search_filters))
+        ext
+      })
+      result$outputID <- result$inputID
+      ui_content <- list()
+      for (i in seq_len(length(df_selectors))) {
+        ui_content[[i * 2]] <- fluidRow(column(1), column(10, DTOutput(ns(paste0("dt_", result$outputID[i])))))
+        ui_content[[i * 2 - 1]] <- fluidRow(column(1), column(10, fileInput(inputId = ns(result$inputID[i]), label = label_pre[[i]],
+                                                                            multiple = TRUE, accept = accept[i])))
+      }
+      tagList(ui_content)
+    } else {
+      paste0("this model has no datafile selectors")
+    }
+  })
+
+  observeEvent(result$outputID, {
+    lapply((seq_len(length(result$outputID))), function(i) {
+      dt_output <- paste0("dt_", result$outputID[[i]])
+      output[[dt_output]] <- renderDT(server = FALSE, {
+        cond <- result$tbl_modelsDetails$data_settings$datafile_selectors[[i]]$allow_multiple
+        if (cond) {
+          select = "multiple"
+        } else {
+          select = "single"
+        }
+        tbl_files_filtered <- result$tbl_files[c(grep(result$outputID[[i]], result$tbl_files$file_description)),]
+        df <- data.frame(id = tbl_files_filtered$id, names = tbl_files_filtered$filename)
+        # show last element of data frame first in table
+        df <- df[nrow(df):1, ]
+        if (length(df) > 0) {
+          colnames(df) <- c("ID", "Name")
+          datatable(df, selection = select, rownames = FALSE)
+        } else {
+          datatable(data.frame("ID" = character(0), "Name" = character(0)), rownames = FALSE)
+        }
+      })
+    })
+  })
+
+  observeEvent(result$inputID, {
+    h <- unlist(result$inputID)
+    df_selectors <- result$tbl_modelsDetails$data_settings$datafile_selectors
+    lapply(seq_len(length(h)), function(i) {
+      check_ext <- paste0(".", unlist(df_selectors[[i]]$search_filters))
+      observeEvent(input[[h[i]]], {
+        for (j in seq_len(nrow(input[[h[i]]]))) {
+          if (paste0(".", tools::file_ext(input[[h[i]]][j,]$name)) %in% check_ext) {
+            .uploadDamageFile(file_entry = h[i], file_name = input[[h[i]]][j, ])
+          } else {
+            oasisuiNotification(type = "error", paste("Extension not supported."))
+          }
```

Comment:
I had some trouble defining a datafile selector in `model_settings.json` that would accept files with any extension. No combination of regular expressions or wildcards worked so I ended up hardcoding a fixed set of extensions. I think the issue is around here or possibly up on line 137.

I also note that the type of search filter is ignored, so `user`, `content_type` or `filename__contains` will all be treated as `filename__extension` (which isn't actually defined in the [schema](https://github.com/OasisLMF/OasisLMF/blob/master/oasislmf/schema/model_settings.json#L954)).

No pressure to fix this now.